tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -276,27 +294,361 @@ test.describe( 'Set up accounts', () => {
+					'Connect to existing Merchant Center account'
+				);
+
+				const mcFooterButton =
+					setUpAccountsPage.getMCCardFooterButton();
+				await expect( mcFooterButton ).toBeEnabled();
+
+				const continueButton = setUpAccountsPage.getContinueButton();
+				await expect( continueButton ).toBeDisabled();
+			} );
+
+			test( 'click "Create account" button should see the modal of confirmation of creating account', async () => {
+				// Click the create account button
+				const createAccountButton =
+					setUpAccountsPage.getMCCreateAccountButtonFromPage();
+				await createAccountButton.click();
+				await page.waitForLoadState( LOAD_STATE.DOM_CONTENT_LOADED );
+
+				const modalHeader = setUpAccountsPage.getModalHeader();
+				await expect( modalHeader ).toContainText(
+					'Create Google Merchant Center Account'
+				);
+
+				const modalCheckbox = setUpAccountsPage.getModalCheckbox();
+				await expect( modalCheckbox ).toBeEnabled();
+
+				const createAccountButtonFromModal =
+					setUpAccountsPage.getMCCreateAccountButtonFromModal();
+				await expect( createAccountButtonFromModal ).toBeDisabled();
+
+				// Click the checkbox of accepting ToS, the create account button will be enabled.
+				await modalCheckbox.click();
+				await expect( createAccountButtonFromModal ).toBeEnabled();
+			} );
+
+			test.describe( 'click "Create account" button from the modal', () => {
+				test( 'should see Merchant Center account "Connected" when the website is not claimed', async () => {
+					await Promise.all( [
+						// Mock Merchant Center create accounts
+						setUpAccountsPage.mockMCCreateAccountWebsiteNotClaimed(),
+
+						// Mock Merchant Center as connected with ID 12345
+						setUpAccountsPage.mockMCConnected( 12345 ),
+					] );
+
+					const createAccountButtonFromModal =
+						setUpAccountsPage.getMCCreateAccountButtonFromModal();
+					await createAccountButtonFromModal.click();
+					await page.waitForLoadState(
+						LOAD_STATE.DOM_CONTENT_LOADED
+					);
+					const mcConnectedLabel =
+						setUpAccountsPage.getGoogleComboConnectedLabel();
+					await expect( mcConnectedLabel ).toContainText(
+						'Connected'
+					);
+
+					const mcDescriptionRow =
+						setUpAccountsPage.getMCDescriptionRow();
+					await expect( mcDescriptionRow ).toContainText(
+						'Merchant Center ID: 12345'
+					);
+				} );
+
+				test.describe( 'when the website is already claimed', () => {
+					test( 'should see "Reclaim my URL" button, "Switch account" button, and site URL input', async ( {
+						baseURL,
+					} ) => {
+						const host = new URL( baseURL ).host;
+
+						await Promise.all( [
+							// Mock Jetpack as connected.
+							setUpAccountsPage.mockJetpackConnected(),
+
+							// Mock google as connected.
+							setUpAccountsPage.mockGoogleConnected(
+								'google@example.com'
+							),
+							setUpAccountsPage.mockAdsAccountsResponse(
+								ADS_ACCOUNTS
+							),
+							setUpAccountsPage.mockMCHasAccounts(),
```

Comment:
These mocks seem to be redundant as the upper `test.beforeAll` blocks have set the same mocks and they won't invalidated when entering here.

https://github.com/woocommerce/google-listings-and-ads/blob/8fba3e4ecd1d8b12f365706545d47f66976a1770/tests/e2e/specs/setup-mc/step-1-accounts.test.js#L297-L304

https://github.com/woocommerce/google-listings-and-ads/blob/8fba3e4ecd1d8b12f365706545d47f66976a1770/tests/e2e/specs/setup-mc/step-1-accounts.test.js#L315-L318

Reply:
Mocks updated.